packages/git/src/browser/git-sync-service.ts

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from "inversify";
+import { MessageService, Emitter, Event } from "@theia/core";
+import { QuickOpenService, QuickOpenItem, QuickOpenMode, ConfirmDialog } from "@theia/core/lib/browser";
+import { GitRepositoryTracker } from "./git-repository-tracker";
+import { Git, Repository, WorkingDirectoryStatus } from "../common";
+
+@injectable()
+export class GitSyncService {
+
+    @inject(Git)
+    protected readonly git: Git;
+
+    @inject(GitRepositoryTracker)
+    protected readonly repositoryTracker: GitRepositoryTracker;
+
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+
+    @inject(QuickOpenService)
+    protected readonly quickOpenService: QuickOpenService;
+
+    protected readonly onDidChangeEmitter = new Emitter<void>();
+    readonly onDidChange: Event<void> = this.onDidChangeEmitter.event;
+
+    protected fireDidChange(): void {
+        this.onDidChangeEmitter.fire(undefined);
+    }
+
+    protected _syncing = false;
+    get syncing(): boolean {
```

Comment:
Please switch to a method from the property getter if possible.

Reply:
I am not sure about this anymore... You know, it is fine that one could extend it, be then we do not use basic language features. It would be strange to have `setSync` `isSync`; you decide.